### PR TITLE
CDI-612 Introduce EventConsumer

### PR DESCRIPTION
**Pull request by mkouba**
- accepts EventContext
- there is no need to define two notifyWith() methods anymore
- allows to throw checked exceptions within a callback

### CODE REVIEW

**Comment by mkouba** [comment]
@cdi-spec/eg please review...

File: api/src/main/java/javax/enterprise/inject/spi/configurator/ObserverMethodConfigurator.java [hunk 1]
```diff
@@ -17,22 +17,22 @@
...
-import java.lang.reflect.Type;
-import java.util.Set;
-import java.util.function.BiConsumer;
-import java.util.function.Consumer;
 
 /**
- * This API is an helper to build a new {@link ObserverMethod} instance.
- * CDI container must provides an implementation of this interface accessible.
+ * This API is an helper to build a new {@link ObserverMethod} instance. CDI container must provides an implementation of this
+ * interface accessible.
```
Comment (tremes) [inline api/src/main/java/javax/enterprise/inject/spi/configurator/ObserverMethodConfigurator.java]: I know this is just formatting change, but this sentence seems to me wrong. I would reword this e.g
`CDI container must provide access to an implementationof this interface.`
This is likely problematic in all other Configurator interfaces.

Comment (mkouba) [inline api/src/main/java/javax/enterprise/inject/spi/configurator/ObserverMethodConfigurator.java]: Yep, at least `must provides` is gramatically wrong. 

I would keep it simple: `The container must provide an implementation of this interface.`?

File: api/src/main/java/javax/enterprise/inject/spi/configurator/ObserverMethodConfigurator.java [hunk 2]
```diff
@@ -149,26 +148,38 @@
...
      *
      * @param callback to call for the event notification
      * @return self
      */
-    ObserverMethodConfigurator<T> notifyWith(Consumer<T> callback);
+    ObserverMethodConfigurator<T> notifyWith(EventConsumer<T> callback);
 
     /**
-     * Defines a BiConsumer to call when event is notified
+     * Allows modification of the asynchronous status of the observer to build.
```
Comment (mkouba) [inline api/src/main/java/javax/enterprise/inject/spi/configurator/ObserverMethodConfigurator.java]: This is also unrelated to CDI-612 (just formatting change,). Pls create a new JIRA issue.

File: spec/src/main/asciidoc/core/spi.asciidoc [hunk 3]
```diff
@@ -1174,8 +1174,6 @@ With `BeanConfigurator` you can perform all the operations defined in <<bean_att
 
 CDI 2.0 introduced the `javax.enterprise.inject.spi.configurator.ObserverMethodConfigurator` interface to help configuring a new `ObserverMethod` instance.
 
-`ObserverMethodConfigurator` must be initialized from a `java.lang.reflect.Method`, `AnnotatedMethod` or an existing `ObserverMethod` with one of its `read()` methods.
-
```
Comment (tremes) [inline spec/src/main/asciidoc/core/spi.asciidoc]: This isn't going to be so easy.  If we merge this then spec won't mention these methods at all.

Comment (mkouba) [inline spec/src/main/asciidoc/core/spi.asciidoc]: Should be fixed now...

**Comment by manovotn** [comment]
+1 from me
